hawkmoth/parser.py

```diff
@@ -415,15 +481,16 @@ def _recursive_parse(domain, comments, errors, cursor, nest):
 
         return [ds]
 
-    elif cursor.kind == CursorKind.FUNCTION_DECL:
-        args = _get_args(cursor)
-        linkage = ' '.join(_get_fn_quals(cursor))
-        ttype = f'{linkage}{" " if linkage else ""}{cursor.result_type.spelling}'
-
+    elif cursor.kind in [CursorKind.FUNCTION_DECL,
+                         CursorKind.CONSTRUCTOR,
+                         CursorKind.DESTRUCTOR,
+                         CursorKind.CXX_METHOD,
+                         CursorKind.FUNCTION_TEMPLATE]:
+        ttype, args, quals = _fn_fixup(cursor)
```

Comment:
I'm thinking maybe we should keep `CursorKind.FUNCTION_DECL` handling separate from the rest, which are all C++ only, and not try to handle them all together. Bundling them leads to a lot of unnecessary stuff in `_fn_fixup()` for regular functions. Similarly, split `_get_fn_quals` to two.

AFAICT `_fn_fixup()` does not need the `CursorKind.FUNCTION_DECL` parts of `_get_fn_quals()`.

Reply:
True, I was trying to keep all the common paths common, but gains are limited. I have a slight prior preference for the current version in this case, but let me see how it looks if I split it.